app/controllers/dashboard_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# DashboardController: Actions associated with presenting dasbhoards
+class DashboardController < ApplicationController
+  include Orchestration::Orchestrator
+
+  before_action :authenticate_user!
+  before_action :authenticate_user_role
+  before_action :set_jurisdiction_paths, :set_all_assigned_users
```

Comment:
let's take the contents of these methods and put them in `dashboard` since `index` is just responsible for redirecting to the correct dashboard, it's unnecessary to set the assigned users and jurisdiction paths there

Reply:
How about if we add an only option, as shown for one below?
`  before_action :authenticate_user!, only: [dashboard]`

Reply:
sure!

Reply:
Restricted action filters to the dashboard method.  [https://github.com/SaraAlert/SaraAlert/pull/1025/commits/b9d5888836c64019559d15bb8b39ea40ccc35fbf](url)

Reply:
ooh, the other actions (authentications) should be for all of the methods though

Reply:
Ok._ I had read your initial comment as a concern about the authentication-related actions happening multiple times when index was invoked.  I was wondering if the current code means authentication isn't checked before the two set_ methods.  Would the following be acceptable? 
```
before_action` :authenticate_user!
before_action :authenticate_user_role
before_action :set_jurisdiction_paths, :set_all_assigned_users, only: %i[dashboard]
```

Reply:
Yeah, my original comment was to move only the `:set_jurisdiction_paths` and `:set_all_assigned_users` methods directly into the `dashboard` method because that's the only method these actions need to occur in. The solution you proposed it acceptable, but I think it would be just simpler to put the contents of both `:set_jurisdiction_paths` and `:set_all_assigned_users` directly into the `dashboard` method